tests/TestCase/Validation/ValidationTest.php

```diff
@@ -1429,6 +1450,7 @@ public function testDateYmNumeric()
      */
     public function testDateY()
     {
+        $this->assertTrue(Validation::date('0001', ['y']));
```

Comment:
Are 3-digit years now supported?

Reply:
They shouldn't be. I can add a test for that though.

Reply:
That also means 2 and 1 digit years then, right?

Reply:
Supporting one, and two digit years gets hard as the 2 digit years are ambiguous with the modern day two digit format (which is a disaster, but still in use). Three digit years should be possible, but I'm not convinced the added complexity is justified given how infrequently these dates will be used.

Reply:
I'm not convinced it's necessary either.  I think it'd be nice to have a failure test case.